Dockerfile

```diff
@@ -1,10 +1,30 @@
-FROM elixir:1.5
-RUN apt-get update -qq && apt-get install -y inotify-tools
-RUN curl -so- -L https://yarnpkg.com/install.sh | bash
-RUN mkdir /ret
-WORKDIR	/ret
-COPY mix.exs /ret/mix.exs
-COPY mix.lock /ret/mix.lock
-RUN mix local.hex --force
-RUN mix local.rebar --force
-COPY . /ret
+FROM elixir:1.8-alpine as builder
+RUN apk add --no-cache nodejs yarn git build-base
+copy . .
+RUN mix local.hex --force && mix local.rebar --force && mix deps.get
```

Comment:
Similar to comments in hubs, I bet we can optimize caching by copying mix.exs first then running `mix deps.get` before copying other things.